tools/moth.py

```diff
@@ -325,18 +316,58 @@ def secrets(self):
 
 class Category:
     def __init__(self, path, seed):
+        """Represents a new category of puzzles.  
+        Gathers the point values for all the puzzles in the category.          
+          1. It pulls the points from the POINTS attribute of path/category.py, if it exists.
+          2. Any path/<integer_dir> is counted as a puzzle as well.
+        When puzzles are generated, they are first built using the completely optional
+        puzzle directory, then passed to the category.make function if it is in 
+        category.POINTS.
+        :param str path: Path to the category directory.
+        :param float seed: The random seed for this categories random number generator."""
+        
         self.path = path
         self.seed = seed
+        
+        if 'category.py' in os.listdir(path):
+            # Good Lord this is forking dangerous.
+            mod_path = os.path.join(path, 'category.py')
+            loader = SourceFileLoader('category_mod', mod_path)
+            category_mod = loader.load_module()
+            if hasattr(category_mod, 'POINTS') and hasattr(category_mod, 'make'):
+                self.category_mod = category_mod
+                self.points.extend([int(p) for p in category_mod.POINTS])
+            else:
+                raise RuntimeError("Trying to use the category module at {}, but it must have a "
+                                   "POINTS sequence and make(puzzle, points) function defined.".format(mod_path))
+        else:
+            self.category_mod = None
+        
         self.pointvals = []
+        if hasattr(category_mod, 'POINTS'):
+            # Points should be a sequence type.
+            self.pointvals.extend(category_mod.POINTS)
+        
         for fpath in glob.glob(os.path.join(path, "[0-9]*")):
-            pn = os.path.basename(fpath)
-            points = int(pn)
+            points = int(os.path.basename(fpath))
             self.pointvals.append(points)
+            
         self.pointvals.sort()
 
     def puzzle(self, points):
-        path = os.path.join(self.path, str(points))
-        return Puzzle(self.seed, path)
+        """Returns the puzzle object for the given point value, or raises a KeyError if no
+        such point value puzzle is defined."""
+        if points not in self.pointvals:
+            raise KeyError("No such puzzle {}".format(points))
```

Comment:
This check bothers me. If we ever encounter a situation where it evaluates true, something's broken in either the devel server or the build script, and this error message won't help the user any more than the traceback it's preventing them from sending to the developers.